eagr/tests/reflection/test_grpc_reflection_interface.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020-present Kensho Technologies, LLC.
+import unittest
+from unittest.mock import MagicMock, patch
+
+from eagr.reflection import grpc_reflection_interface, reflection_descriptor_database
+from eagr.tests.reflection import utils
+
+
+# Mocks to patch out context-specific method calls in eagr.reflection.grpc_reflection_interface
+def build_database_from_channel_mock(channel):
+    """Returns a built DescriptorPool from a mock ServerReflectionStub, given any channel"""
+    return reflection_descriptor_database.build_database_from_stub(utils.reflection_client_mock)
+
+
+def find_service_by_name_mock(service_name):
+    """Returns a mock GRPC service given a valid service name"""
+    method_mock = MagicMock()
+    method_mock.name = "MyTestMethod"
+    method_mock.input_type = MagicMock()
+    method_mock.input_type.prototype = build_database_from_channel_mock
+
+    service_mock = MagicMock()
+    service_mock.methods = [method_mock]
+    if service_name == "good_service":
+        return service_mock
+
+    else:
+        raise KeyError("Bad service {} given".format(service_name))
+
+
+def make_grpc_unary_method_mock(channel, service_name, method_descriptor, symbol_database_instance):
+    """Returns a mock callable method regardless of params provided"""
+
+    def callable_mock():
+        """Return a mock string value invariably"""
+        return "return_value"
+
+    return callable_mock
+
+
+def get_prototype_mock(input_type):
+    """Returns a mock callable method regardless of params provided"""
+
+    def callable_mock():
+        """Return a mock string value invariably"""
+        return "return_value"
+
+    return callable_mock
```

Comment:
You should consider type-hinting test functionality as well, that will make mypy more effective. Tests have the largest surface area touching your code, and if all of that surface area is typed `Any` then it's not much use.

Reply:
I agree--I do plan on adding type hinting throughout in a future PR. Right now only certain modules have it.